### PR TITLE
Update CSM-Replication module

### PR DESCRIPTION
**Pull request by chimanjain**
# Description
Update CSM-Replication module

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Built Cert-csi

### CODE REVIEW

**Review by adarsh-dell — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -2,12 +2,10 @@ module github.com/dell/cert-csi
 
 go 1.22.0
 
-toolchain go1.22.5
```
Comment (adarsh-dell) [inline go.mod:5]: Any impact from removing this toolchain?

Comment (adarsh-dell) [reply go.mod:5]: I hope you have tried to build the binary and use it for some basic sanity before raising the PR.

Comment (chimanjain) [reply go.mod:5]: No, if we don't mention toolchain, it will take the default go version present in our machine.

Comment (chimanjain) [reply go.mod:5]: I have build the binary successfully.

Comment (adarsh-dell) [reply go.mod:5]: I can see https://github.com/dell/cert-csi/blob/0233a8536958961763bc4d2e1b5dd5e320d6f7f0/Makefile#L55 so for docker image it should be fine.
Thanks for the confirmation.

**Review by AkshaySainiDell — APPROVED** [review]

**Pull request merged**